head/src/main/kotlin/io/qalipsis/core/head/configuration/ZoneServiceImpl.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * QALIPSIS
+ * Copyright (C) 2025 AERIS IT Solutions GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package io.qalipsis.core.head.configuration
+
+import io.qalipsis.core.head.jdbc.repository.ZoneRepository
+import io.qalipsis.core.head.model.Zone
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.toList
+
+/**
+ * Default implementation of the [ZoneService] interface.
+ *
+ * @author Francisca Eze
+ */
+internal class ZoneServiceImpl(private val zoneRepository: ZoneRepository): ZoneService {
```

Comment:
You need an implementation when the persistence is activated (this one actually), and one when ExecutionEnvironments.TRANSIENT is active. Which always returns an empty list. Otherwise, QALIPSIS will not be able to run when ExecutionEnvironments.TRANSIENT is active because the ZoneController will not be able to receive any instance of ZoneService.